api/v1alpha1/addresspool_webhook_test.go

```diff
@@ -0,0 +1,123 @@
+package v1alpha1
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	MetalLBTestNameSpace = "metallb-test-namespace"
+)
+
+func TestAddressPoolWebhook(t *testing.T) {
+	autoAssign := false
+	addressPool := AddressPool{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-addresspool",
+			Namespace: MetalLBTestNameSpace,
+		},
+		Spec: AddressPoolSpec{
+			Protocol: "layer2",
+			Addresses: []string{
+				"1.1.1.1-1.1.1.100",
+			},
+			AutoAssign: &autoAssign,
+		},
+	}
+	addressPoolList := &AddressPoolList{}
+	addressPoolList.Items = append(addressPoolList.Items, addressPool)
+
+	tests := []struct {
+		desc             string
+		addressPool      *AddressPool
+		isNewAddressPool bool
+		expectedError    string
+	}{
+		{
+			desc: "Second AddressPool, already defined name",
```

Comment:
Can we also check what happens if the first item of the range is after the second?
Like 1.1.1.200-1.1.1.101